apps/meteor/app/push/server/apn.ts

```diff
@@ -1,22 +1,46 @@
+import type { IAppsTokens, RequiredField } from '@rocket.chat/core-typings';
 import apn from 'apn';
 import EJSON from 'ejson';
 
+import type { PushOptions, PendingPushNotification } from './definition';
 import { logger } from './logger';
 
-let apnConnection;
-
-export const sendAPN = ({ userToken, notification, _removeToken }) => {
-	if (typeof notification.apn === 'object') {
-		notification = Object.assign({}, notification, notification.apn);
+let apnConnection: apn.Provider | undefined;
+
+// the definition included in the module does not include the set-only attributes
+class ApnNotification extends apn.Notification {
+	setContentAvailable: (value: boolean | 1 | 0) => void;
+
+	// eslint-disable-next-line @typescript-eslint/no-empty-function
+	set category(_value: string | undefined) {}
+
+	// eslint-disable-next-line @typescript-eslint/no-empty-function
+	set body(_value: string) {}
+
+	// eslint-disable-next-line @typescript-eslint/no-empty-function
+	set title(_value: string) {}
+}
+
+export const sendAPN = ({
+	userToken,
+	notification,
+	_removeToken,
+}: {
+	userToken: string;
+	notification: PendingPushNotification & { topic: string };
+	_removeToken: (token: IAppsTokens['token']) => void;
+}) => {
+	if (!apnConnection) {
+		throw new Error('Apn Connection not initialized.');
 	}
 
 	const priority = notification.priority || notification.priority === 0 ? notification.priority : 10;
 
-	const note = new apn.Notification();
+	const note = new apn.Notification() as ApnNotification;
 
 	note.expiry = Math.floor(Date.now() / 1000) + 3600; // Expires 1 hour from now.
-	note.badge = notification.badge;
-	note.sound = notification.sound;
+	note.badge = notification.badge as number; // the module accepts undefined, but is typed wrong
+	note.sound = notification.sound as string; // the module accepts undefined, but is typed wrong
```

Comment:
even if the type is incorrect, we can make it work:
```suggestion
	if (notification.badge) {
		note.badge = notification.badge;
	}
	if (notification.sound) {
		note.sound = notification.sound;
	}
```

Reply:
I had done this at first, but ultimately decided to go with the typecast instead, to ensure that undefined values will continue to be set to the attribute.